src/test/java/net/minecraftforge/debug/block/TrapdoorTest.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2019.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.debug.block;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.LadderBlock;
+import net.minecraft.block.SoundType;
+import net.minecraft.block.material.Material;
+import net.minecraft.client.renderer.RenderType;
+import net.minecraft.client.renderer.RenderTypeLookup;
+import net.minecraft.item.BlockItem;
+import net.minecraft.item.Item;
+import net.minecraftforge.event.RegistryEvent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.common.Mod.EventBusSubscriber.Bus;
+import net.minecraftforge.fml.event.lifecycle.FMLClientSetupEvent;
+
+@SuppressWarnings("unused")
+@Mod(TrapdoorTest.MODID)
+public class TrapdoorTest
+{
+    public static final String MODID = "trapdoor_test";
+
+    // Represents modded ladders to be supported
+    private static class TrapdoorTestLadder extends LadderBlock
+    {
+        private TrapdoorTestLadder(Block.Properties builder)
+        { super(builder); }
+    }
+
+    private static final Block MODDED_TEST_LADDER = (new TrapdoorTestLadder(
+      // Using different block properties as the model is the vanilla ladder model.
```

Comment:
No, static creation of registry objects is a BIG NO

Reply:
applied